verifiers/internal/gha/builder.go

```diff
@@ -175,39 +190,309 @@ func verifyTrustedBuilderRef(id *WorkflowIdentity, ref string) error {
+	HostedGitHub
+)
+
+// See https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md.
 type WorkflowIdentity struct {
-	// The caller repository
-	CallerRepository string `json:"caller"`
-	// The commit SHA where the workflow was triggered
-	CallerHash string `json:"commit"`
-	// Current workflow (reuseable workflow) ref
-	JobWobWorkflowRef string `json:"job_workflow_ref"`
-	// Trigger
-	Trigger string `json:"trigger"`
+	// The source repository
+	SourceRepository string
+	// The commit SHA where the workflow was triggered.
+	SourceSha1 string
+	// Ref of the source.
+	SourceRef *string
+	// ID of the source repository.
+	SourceID *string
+	//  Source owner ID of repository.
+	SourceOwnerID *string
+
+	// Workflow path OIDC subject - ref of reuseable workflow or trigger workflow.
+	SubjectWorkflowRef string
+	// Subject commit sha1.
+	SubjectSha1 *string
+	// Hosted status of the subject.
+	SubjectHosted *Hosted
+
+	// BuildTrigger
+	BuildTrigger string
+	// Build config path, i.e. the trigger workflow.
+	BuildConfigPath *string
+
+	// Run ID
+	RunID *string
 	// Issuer
-	Issuer string `json:"issuer"`
+	Issuer string
+}
+
+func getHosted(cert *x509.Certificate) (*Hosted, error) {
+	runnerEnv, err := getExtension(cert, "1.3.6.1.4.1.57264.1.11", true)
+	if err != nil {
+		return nil, err
+	}
+	if runnerEnv == "github-hosted" {
+		r := HostedGitHub
+		return &r, nil
+	}
+	if runnerEnv == "self-hosted" {
+		r := HostedSelf
+		return &r, nil
+	}
+	return nil, nil
+}
+
+func validateClaimsEqual(deprecated, existing string) error {
+	// derecated may be empty, but it more likely the cert is old and 'new' is empty.
+	if deprecated != "" && existing != "" && deprecated != existing {
+		return fmt.Errorf("%w: '%v' != '%v'", serrors.ErrorInvalidFormat, deprecated, existing)
+	}
+	if deprecated == "" && existing == "" {
+		return fmt.Errorf("%w: claims are empty", serrors.ErrorInvalidFormat)
+	}
+	return nil
 }
 
 // GetWorkflowFromCertificate gets the workflow identity from the Fulcio authenticated content.
+// See https://github.com/sigstore/fulcio/blob/e763d76e3f7786b52db4b27ab87dc446da24895a/pkg/certificate/extensions.go.
+//
+//nolint:gocyclo
 func GetWorkflowInfoFromCertificate(cert *x509.Certificate) (*WorkflowIdentity, error) {
 	if len(cert.URIs) == 0 {
-		return nil, errors.New("missing URI information from certificate")
+		return nil, fmt.Errorf("%w: missing URI information from certificate", serrors.ErrorInvalidFormat)
+	}
+
+	// 1.3.6.1.4.1.57264.1.2: DEPRECATED.
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#1361415726412--github-workflow-BuildTrigger-deprecated
+	deprecatedBuildTrigger, err := getExtension(cert, "1.3.6.1.4.1.57264.1.2", false)
+	if err != nil {
+		return nil, err
+	}
+	// 1.3.6.1.4.1.57264.1.20 | Build Trigger
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264120--build-trigger
+	buildTrigger, err := getExtension(cert, "1.3.6.1.4.1.57264.1.20", true)
```

Comment:
nit: Could you just get the string of the `ObjectIdentifier`?

```suggestion
	buildTrigger, err := getExtension(cert, OIDBuildTrigger.String(), true)
```